src/main/kotlin/org/malachite/estella/offer/api/OfferController.kt

```diff
@@ -29,50 +29,47 @@ class OfferController(
     fun getOffers(): ResponseEntity<List<OfferResponse>> =
         offerService.getOffers()
             .map { it.toOfferResponse() }
-            .let { ResponseEntity(it, HttpStatus.OK) }
+            .let { ResponseEntity.ok(it) }
 
 
     @CrossOrigin
     @GetMapping("/{offerId}")
     fun getOffer(@PathVariable offerId: Int): ResponseEntity<OfferResponse> =
         offerService.getOffer(offerId)
-            .let { ResponseEntity(it.toOfferResponse(), HttpStatus.OK) }
+            .let { ResponseEntity.ok(it.toOfferResponse()) }
 
     @CrossOrigin
     @PostMapping()
     fun addOffer(
         @RequestHeader(EStellaHeaders.jwtToken) jwt: String?,
         @RequestBody offer: OfferRequest
-    ): ResponseEntity<Message> {
-        val hrPartner = getHrPartnerFromJWT(jwt)
-        offerService.addOffer(offer, hrPartner)
-        return ResponseEntity(SuccessMessage, HttpStatus.OK)
-    }
+    ): ResponseEntity<OfferResponse> =
+        getHrPartnerFromJWT(jwt)
+            .let { offerService.addOffer(offer, it) }
+            .let(Offer::toOfferResponse)
+            .let { OwnResponses.CREATED(it) }
 
     @CrossOrigin
     @PutMapping("/{offerId}")
     fun updateOffer(
         @RequestHeader(EStellaHeaders.jwtToken) jwt: String?,
         @PathVariable("offerId") offerId: Int,
         @RequestBody offer: OfferRequest
-    ): ResponseEntity<Any> {
-        val hrPartner = getHrPartnerFromJWT(jwt)
-        offerService.updateOffer(offerId, offer, hrPartner)
-        return ResponseEntity(SuccessMessage, HttpStatus.OK)
-    }
+    ): ResponseEntity<Any> =
+        getHrPartnerFromJWT(jwt)
+            .let { offerService.updateOffer(offerId, offer, it) }
```

Comment:
Zmieniłbym tu nazwę z `offer` na `offerRequest`, bo to wciąż jednak request jest

Reply:
trzeba by zerknąć na inne kontrolery bo wydaje mi się że tu też nie ma żadnej konwencji